modules/pel/peltool/parse_user_data.py

```diff
@@ -40,25 +40,36 @@ def parse(self) -> str:
         else:
             value = self.parseCustom()
 
+        # Catch if value is None, otherwise python crashes
+        if value == None:
+            value = ("Parser returned a value of None for creatorID={} compID={} subType={} version={}"
+                     .format(self.creatorID, hex(self.compID), self.subType, self.version))
+            return json.dumps(value)
         return value
 
     def parseCustom(self) -> str:
         import importlib
         name = (self.creatorID.lower() + "%04X" % self.compID).lower()
         try:
             cls = importlib.import_module("udparsers." + name + "." + name)
-            mv = memoryview(self.data)
-            return cls.parseUDToJson(self.subType, self.version, mv)
+            if self.data:
+                mv = memoryview(self.data)
+                return cls.parseUDToJson(self.subType, self.version, mv)
         except ImportError:
-            mv = memoryview(self.data)
-            return json.dumps(hexdump(mv))
+            # No print for informational purposes, this is encountered often, e.g. PHYP
+            if self.data:
+                mv = memoryview(self.data)
+                return json.dumps(hexdump(mv))
         except Exception as e:
-            print('Failed parsing user data for creator {} compID {} '
-                  'subType {} version {}: {}'.format(
-                      self.creatorID, "0x%04X" % self.compID,
-                      "0x%X" % self.subType, self.version, str(e)), file=sys.stderr)
-            mv = memoryview(self.data)
-            return json.dumps(hexdump(mv))
+            d = dict()
+            # in case we do NOT have data, dump the Error at a minimum
+            if self.data:
+                mv = memoryview(self.data)
+                d["Data"] = hexdump(mv)
+            d["Error"] = ("Failed parsing user data for creator={} compID={} subType={} version={} Exception={}"
```

Comment:
At first this looked like an indentation error to me though it isn't.  could you move it up after line 64 to make it more obvious it should always run?

Reply:
Will do.